tests/tests/test_provides_depends.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/python
+# Copyright 2020 Northern.tech AS
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        https://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+import subprocess
+import time
+
+from ..common_setup import enterprise_no_client
+from .common_update import update_image, common_update_procedure
+from .mendertesting import MenderTesting
+from ..MenderAPI import auth, auth_v2, deploy, image, logger
+
+from testutils.infra.device import MenderDevice
+
+
+class TestProvidesDependsEnterprise(MenderTesting):
+    def test_update_provides_depends(self, enterprise_no_client):
+        """
+        Perform two consecutive updates, the first adds virtual provides
+        to the artifact and the second artifact depends on these provides.
+        """
+
+        DEMO_POLL_INTERVAL = 5
+        IMAGE_NAME = "core-image-full-cmdline-qemux86-64.ext4"
+
+        # Create tenant user
+        auth.reset_auth_token()
+        auth.new_tenant("admin", "bob@builder.org", "secret-service")
+        token = auth.current_tenant["tenant_token"]
+
+        # Create client setup with tenant token
+        enterprise_no_client.new_tenant_docker_client("mender-client", token)
+        mender_device = MenderDevice(enterprise_no_client.get_mender_clients()[0])
+        mender_device.host_ip = enterprise_no_client.get_virtual_network_host_ip()
```

Comment:
I think `host_ip` does not belong go `mender_device`. That is why when I introduced this in all other tests, I always kept it independently. Better to be consistent with what we have in other tests.

Reply:
But this is the IP of the device, no? Not sure if I understand the reason.

Reply:
No, is the IP of the test runner (i.e. host running docker-compose) as seen from the device (i.e. gateway). At the time, there was a naming discussion around `get_virtual_network_host_ip` due to this.

Reply:
The IP of the device(s) comes from `get_mender_clients()`

Reply:
The `host_ip` is just a variable with the client's host container IP that I added to the `mender_device` object as a matter of keeping the variable associated with the device. I don't think it's defined anywhere else, and might as well be defined as it's own variable.

Reply:
I see, thanks for explaining. Fixed!